llvm/test/Transforms/DFAJumpThreading/dfa-jump-threading-transform.ll

```diff
@@ -300,3 +300,237 @@ define void @self-reference() {
 end:
   ret void
 }
+
+@a = external dso_local global ptr, align 8
+@b = external dso_local global i32, align 4
+@c = external dso_local global i64, align 8
+@d = external dso_local global i16, align 2
+@e = external dso_local global i64, align 8
+@f = external dso_local global i32, align 4
+@g = external dso_local global i64, align 8
+@h = external dso_local global i32, align 4
+@i = external dso_local global ptr, align 8
+
+define void @pr106083_invalidBBarg_fold() {
```

Comment:
Can it be further reduced? I feel a bit awkward understanding the bugs.

Reply:
Tools like `llvm-reduce` may help.

Reply:
done